tests/harness/Test/Parse.hs

```diff
@@ -227,9 +227,9 @@ results :: TestGroupData -> Parser (Either ResultsException [ModuleInfo])
 results TestGroupData {..} =
   (Left (UpToDateException tgdName) <$ upToDate)
   <|> (do
-          skipStartupTrash
-          xs <- P.many moduleInfo
-          skipEndingTrash
+          xs <- skipStartupTrash *> P.many moduleInfo <* skipEndingTrash
+          -- Benchmarks might run twice
+          _ys <- P.option [] $ P.many moduleInfo <* skipEndingTrash
```

Comment:
I am confused as to what this does and why it's necessary.  Why are the benchmarks run twice?  When are they run twice?  Why can we discard the second run?

Reply:
I think `cabal` is calling `ghc` twice sometimes. Thus, if I use `-fforce-recomp` ghc produces sometimes twice the output of compilation, which is what I'm parsing here.